src/composed/key/builder.rs

```diff
@@ -6,11 +6,12 @@ use rand::{thread_rng, CryptoRng, Rng};
 use smallvec::SmallVec;
 
 use crate::composed::{KeyDetails, SecretKey, SecretSubkey};
+use crate::crypto::aead::AeadAlgorithm;
```

Comment:
can you expand the tests in this file please to cover all current versions & algorithms please

Reply:
good point, thanks. i've tried to cover all new variants in 7cd74f89bbeab68b6aafac666f2ea855040524f7.

we can't currently do the RSA test (key locking isn't fully in this PR, for v6 keys), i'd  just leave that as a TODO until the followup PR that implements the new encryption mechanisms (including "v6 key locking").

orthogonally: it occurred to me that it would be nice if these tests also did an encrypt-decrypt roundtrip with the encryption subkey (if you agree, i'd propose i open a ticket about that, and do that outside of this PR).

Reply:
yes, sounds good

Reply:
Opened https://github.com/rpgp/rpgp/issues/384 for encryption roundtripping, later